### PR TITLE
Feature/poi info

### PR DESCRIPTION
**Pull request by tberkane**
Adds a PoiActivity, which can be launched by a button in the main activity.
Launching by clicking on a poi on the map will come in a later pr.

### CODE REVIEW

File: app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    private fun getOnSuccessListener(
+        placeId: String,
+        view: View,
+        placesClient: PlacesClient
+    ): (FetchPlaceResponse) -> Unit =
+        { response: FetchPlaceResponse ->
+            val place = response.place
+            Log.i(TAG, placeId)
```
Comment (aman3014) [inline app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt:66]: is this leftover from debugging

**Review by aman3014 — APPROVED** [review]

File: app/src/main/java/com/github/epfl/meili/poi/PoiActivity.kt [hunk 2]
```diff
@@ -3,52 +3,54 @@ package com.github.epfl.meili.poi
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
 import androidx.fragment.app.Fragment
-import androidx.fragment.app.FragmentActivity
 import androidx.viewpager2.adapter.FragmentStateAdapter
 import androidx.viewpager2.widget.ViewPager2
 import com.github.epfl.meili.R
 import com.github.epfl.meili.map.MapActivity
+import com.google.android.gms.maps.model.PointOfInterest
```
Comment (ahmedc3ca) [inline app/src/main/java/com/github/epfl/meili/poi/PoiActivity.kt:10]: I'm still crashing when clicking from the map view because of this discontinuity of POI versions I think.

**Review by ahmedc3ca — COMMENTED** [review]
Looks very nice! I just can't demo from the map view. Also, for me, the main page is a bit wonky (the buttons are stacked on top of each other).

File: app/src/main/java/com/github/epfl/meili/map/MapActivity.kt [hunk 3]
```diff
@@ -99,14 +99,11 @@ class MapActivity : AppCompatActivity(), OnMapReadyCallback {
 
         // Add on click listener
         clusterManager.setOnClusterItemClickListener {
-            val intent = Intent(this, PoiActivity::class.java)
```
Comment (MarcelTorne) [inline app/src/main/java/com/github/epfl/meili/map/MapActivity.kt:102]: Why did you remove this part? This was supposed to launch the PoiInfo activity when you click on a Poi

**Review by MarcelTorne — COMMENTED** [review]
Seems really good to me! Thanks for the work! Left one small comment.

File: app/src/main/java/com/github/epfl/meili/map/MapActivity.kt (hunk 3 above)
Comment (tberkane) [reply app/src/main/java/com/github/epfl/meili/map/MapActivity.kt:102]: I removed it because the query made in the poi info activity was not working (it didn't return anything). So I think that something doesn't work in the PointOfInterest which is passed in the intent by the MapActivity.

File: app/src/main/java/com/github/epfl/meili/poi/PoiActivity.kt (hunk 2 above)
Comment (tberkane) [reply app/src/main/java/com/github/epfl/meili/poi/PoiActivity.kt]: It should be fixed now (I removed clicking on pois for now).

File: app/build.gradle [hunk 4]
```diff
@@ -69,10 +70,11 @@ dependencies {
     implementation "androidx.activity:activity-ktx:1.2.2"
     implementation "androidx.fragment:fragment-ktx:1.3.2"
     implementation "androidx.viewpager2:viewpager2:1.0.0"
-
+    implementation 'com.google.android.material:material:1.3.0'
+    implementation "androidx.viewpager2:viewpager2:1.0.0"
```
Comment (aman3014) [inline app/build.gradle:74]: These are duplicate dependencies

File: app/src/androidTest/java/com/github/epfl/meili/poi/PoiActivityTest.kt [hunk 5]
```diff
@@ -1,24 +1,87 @@
...
+        `when`(mockPlacesClient.fetchPhoto(any())).thenReturn(tcs2.task)
+
+        `when`(mockPlaces.getPlacesClient(any(), any())).thenReturn(mockPlacesClient)
+
+        PoiInfoFragment.placesClientService = { mockPlaces }
     }
 
+    @get:Rule
+    val mActivityTestRule: ActivityTestRule<PoiActivity> =
+        object : ActivityTestRule<PoiActivity>(PoiActivity::class.java) {
```
Comment (aman3014) [inline app/src/androidTest/java/com/github/epfl/meili/poi/PoiActivityTest.kt:71]: Please use ActivityScenarioRule, ActivityTestRule is deprecated

**Review by aman3014 — CHANGES_REQUESTED** [review]
Thanks for the work! Couple of minor changes requested

File: app/src/main/java/com/github/epfl/meili/map/MapActivity.kt (hunk 3 above)
Comment (MarcelTorne) [reply app/src/main/java/com/github/epfl/meili/map/MapActivity.kt:102]: I see that's probably because the uid of the poi is from the overpass api and not from Google Places API. This will be fixed in my next PR. I think we should still keep it and the activity should be failure-safe and handle the case when the response is empty. What do you think?

File: app/src/androidTest/java/com/github/epfl/meili/poi/PoiActivityTest.kt (hunk 5 above)
Comment (tberkane) [reply app/src/androidTest/java/com/github/epfl/meili/poi/PoiActivityTest.kt:71]: Ok

File: app/src/main/java/com/github/epfl/meili/map/MapActivity.kt (hunk 3 above)
Comment (tberkane) [reply app/src/main/java/com/github/epfl/meili/map/MapActivity.kt]: Yes, that's better. It's done now.

Comment (MarcelTorne) [reply app/src/main/java/com/github/epfl/meili/map/MapActivity.kt]: Perfect! Thanks for the work

**Review by MarcelTorne — APPROVED** [review]

**Review by aman3014 — APPROVED** [review]
